scratch/copy-mk.py

```diff
@@ -111,48 +111,81 @@ async def async_main(args) -> int:
+    band = next((k for k, v in BANDS.items() if v.adc_sample_rate == adc_sample_rate), None)
     if not band:
         print(
-            f"Unable to determine which band. Sample rate reported as {sample_rate}",
+            f"Unable to determine which band. Sample rate reported as {adc_sample_rate}",
             file=sys.stderr,
         )
         return 1
 
-    out_kwargs = {
-        "name": target.name.replace(".", "_"),
-        "antennas": target.n_antennas,
-        "channels": channels,
-        "digitiser-address": mcast_groups[0].split("+")[0],
-        "sync-time": sync_time,
-        "band": band,
-        "develop": args.develop,
-        # We don't want the dummy antennas for sim_correlator.py
-        "input-labels": ",".join(input_labels[: 2 * target.n_antennas]),
+    config: dict = {
+        "version": "4.5",
+        "config": {},
+        "inputs": {
+            label: {
+                "type": "dig.baseband_voltage",
+                "sync_time": sync_time,
+                "band": band,
+                "adc_sample_rate": adc_sample_rate,
+                "centre_frequency": BANDS[band].centre_frequency,  # TODO get from CAM
+                "antenna": label[:-1],
+                "url": f"spead://{addr}",
+            }
+            for label, addr in zip(input_labels, mcast_groups)
+        },
+        "outputs": {
+            "antenna-channelised-voltage": {
+                "type": "gpucbf.antenna_channelised_voltage",
+                "src_streams": input_labels,
+                "input_labels": input_labels,
+                "n_chans": channels,
+            },
+            "baseline-correlation-products": {
+                "type": "gpucbf.baseline_correlation_products",
+                "src_streams": ["antenna-channelised-voltage"],
+                "int_time": int_time,
+            },
+        },
     }
 
-    out_cmd = [os.path.join(os.path.dirname(__file__), "sim_correlator.py")]
-    out_cmd += [f"--{k}={v}" for k, v in out_kwargs.items() if v is not None]
-
-    if args.image_override is not None:
-        for override in args.image_override:
-            out_cmd.append(f"--image-override={override}")
-
-    out_cmd.append(args.controller)
-
-    print(f"Executing: {out_cmd}\n")
-    os.execv(out_cmd[0], out_cmd)
+    katgpucbf.configure_tools.apply_arguments(config, args)
+
+    if args.dry_run:
+        json.dump(config, sys.stdout, indent=2)
+    else:
+        client = await aiokatcp.Client.connect(args.controller.host, args.controller.port)
+        try:
+            reply, _ = await client.request("product-configure", target.name.replace(".", "_"), json.dumps(config))
+            pc_host = aiokatcp.decode(str, reply[1])
+            pc_port = aiokatcp.decode(int, reply[2])
+            print(f"Product controller is at {pc_host}:{pc_port}")
+            client.close()
+            await client.wait_closed()
+            client = await aiokatcp.Client.connect(pc_host, pc_port)
```

Comment:
Perhaps a bit pedantic, but reading this, do you think it's worth updating the cmd-line arg `--controller` to `--master-controller`?
* I know it's scratch script, and
* I guess you've ~avoided ambiguity by calling this `pc_host` and `pc_port`

Not important, but let me know what you think.